core/src/test/java/io/grpc/internal/RetriableStreamTest.java

```diff
@@ -1506,6 +1508,30 @@ public void normalRetry_thenNoTransparentRetry_andNoMoreRetry() {
     verify(retriableStreamRecorder).postCommit();
   }
 
+  @Test
+  public void droppedShouldNeverRetry() {
+    ClientStream mockStream1 = mock(ClientStream.class);
+    ClientStream mockStream2 = mock(ClientStream.class);
+
+    // start
+    doReturn(mockStream1).when(retriableStreamRecorder).newSubstream(0);
+    retriableStream.start(masterListener);
+
+    verify(retriableStreamRecorder).newSubstream(0);
+    ArgumentCaptor<ClientStreamListener> sublistenerCaptor1 =
+        ArgumentCaptor.forClass(ClientStreamListener.class);
+    verify(mockStream1).start(sublistenerCaptor1.capture());
+
+    // normal retry
+    doReturn(mockStream2).when(retriableStreamRecorder).newSubstream(1);
```

Comment:
All stubbing should happen at the beginning of the test. You don't know whether the stubbed method was called before this line.

Reply:
Done.